mono/utils/mono-mmap.h

```diff
@@ -59,10 +59,11 @@ MONO_API void* mono_valloc     (void *addr, size_t length, int flags, MonoMemAcc
 MONO_API void* mono_valloc_aligned (size_t length, size_t alignment, int flags, MonoMemAccountType type);
 MONO_API int   mono_vfree      (void *addr, size_t length, MonoMemAccountType type);
 MONO_API void* mono_file_map   (size_t length, int flags, int fd, guint64 offset, void **ret_handle);
+
 // Last two parameters are optional.
 // This is mono_file_map but with optionally returning an error message.
 // See https://github.com/mono/mono/issues/8225.
-#ifdef HOST_WIN32
+#if defined (HOST_WIN32) || defined (__cplusplus) // FIXMEcxx is this needed? G_BEGIN_DECLS instead?
```

Comment:
It seems like (maybe it was broken) that we didn't need `MONO_API` previously. And I assume that, since you are touching it, it is broken for the C++ case. `G_EXTERN_C` then seems like the necessary minimum.

Reply:
MONO_API I think is needed for C on Windows.
Because here:
https://jenkins.mono-project.com/job/x/11221/parsed_console/log.html
```
main.obj : error LNK2019: unresolved external symbol mono_file_map_error referenced in function probe_embedded [d:\j\workspace\x\msvc\mono.vcxproj]
```

and then I added it and the error went away.

`MONO_API` is likely overkill for Unix/C++ agreed.
It would take me a while to confirm why `G_EXTERN_C` is needed for Unix/C++ but I believe it is. If you want me to be certain, which I can understand, I can remove the change from this PR and run another CI as C++.